jlm/util/Worklist.hpp

```diff
@@ -294,6 +294,62 @@ class TwoPhaseLrfWorklist final : public Worklist<T>
   std::unordered_map<T, size_t> LastFire_;
 };
 
+/**
+ * A fake worklist that only holds a single bit of information:
+ *  "Has any item been pushed since the last reset?"
+ * Used to implement the Topological worklist policy, which is not technically a worklist policy
+ * @tparam T the type of the work items.
+ * @see Worklist
+ */
+template<typename T>
+class DummyWorklist final : public Worklist<T>
+{
+public:
+  ~DummyWorklist() override = default;
+
+  DummyWorklist() = default;
+
+  bool
+  HasMoreWorkItems() override
```

Comment:
The name indicates that this should also be `const` maybe even `noexcept`, but the base class does not have it (and there might be a reason why it is not).